src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/utils.ts

```diff
@@ -16,31 +13,23 @@ import {
   PVC_SOURCE_NAME,
   REGISTRY_SOURCE_NAME,
   SOURCE_OPTIONS_IDS,
-} from './constants';
+  UPLOAD_SOURCE_NAME,
+} from '../constants';
 
 export const getSourceTypeFromDiskSource = (
   diskSource: V1beta1DataVolumeSpec | V1ContainerDiskSource,
 ): SOURCE_OPTIONS_IDS => {
-  if ((diskSource as V1ContainerDiskSource).image) return CONTAINER_DISK_SOURCE_NAME;
+  if ('image' in diskSource) return CONTAINER_DISK_SOURCE_NAME;
 
-  if ((diskSource as V1beta1DataVolumeSpec).source.blank) return BLANK_SOURCE_NAME;
-  if ((diskSource as V1beta1DataVolumeSpec).source.http) return HTTP_SOURCE_NAME;
-  if ((diskSource as V1beta1DataVolumeSpec).source.registry) return REGISTRY_SOURCE_NAME;
-  if ((diskSource as V1beta1DataVolumeSpec).source.pvc) return PVC_SOURCE_NAME;
+  if ('source' in diskSource && diskSource.source?.blank) return BLANK_SOURCE_NAME;
```

Comment:
why source in diskSource? simply diskSource?.source?.blank is ok

Reply:
same go to all below

Reply:
so you can make XXX = diskSource as YYY and ZZZ = diskSoruce as AAA and void all the unneeded checks.